### ISSUE TITLE
QnA wont get any content from Kendra , if the Kendra content is not in english. 

### ISSUE THREAD

**Issue by Guillaume-Bourque-Levio** [aws-solutions/qnabot-on-aws#713]
**Describe the bug**
From the default QnA client only English content will be replied by QnA bot.

Even if the QnA langage is set to French, and that we have content in French in Kendra, no data is return to my french question.

As soon as I add a data source in English to Kendra I see only those answer in the chat bot.

**To Reproduce**
Create de default QnA bot and select French as language

Create a kendra index and add a web crawler with data in french.

**Expected behavior**
French response available in french that we can see from the kendra console if we set the language to Frnech

**Please complete the following information about the solution:**
- [ ] Version: [e.g. v5.5.1]

- [ ] Region: ca-central-1
- [ ] Was the solution modified from the version published on this repository?
 No

- [ ] Have you checked your [service quotas](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for the services this solution uses?   N/A
- [ ] Were there any errors in the CloudWatch Logs?
No

- I was told that the kendra call must include a language_code  something like this.

```
response = kendra.retrieve(
PageSize = 5,
PageNumber = 1,
QueryText = text,
IndexId = index_id,
AttributeFilter={ "EqualsTo": {
"Key": "_language_code",
"Value": {
"StringValue": "fr"
}
}
```

- Another point can we passe more than 1 language to kendra.retrieve ?

 TIA

Guillaume.

**Comment by dougtoppin** [comment]
@Guillaume-Bourque-Levio thanks for your report, we will investigate and get back to you

**Comment by bios6** [comment]
Hi @Guillaume-Bourque-Levio ,

Please try the following and confirm: For this go to the AWS Console and search for Kendra, then look for the Data Management on the left hand side of the AWS console and click on ```Search indexed content``` . Then try query your question there that you are asking QnABot (in French). If it is not retrieving anything then it's a issue on the Kendra side. If it is indeed an issue on the Kendra side then try to set the Kendra data sources language to be English (which is the default) and you can still pass in the French data sources and you should be able to query in French.

Thanks!

**Comment by Guillaume-Bourque-Levio** [comment]
Hello,

if I try to search in the text box I get nothing because the search only search in the english documents and my document are in french.

But if I change then kendra search behavior to French as you see here kendra will return information from my French documents.

So from what I understand Kendra is fine

![image](https://github.com/aws-solutions/qnabot-on-aws/assets/165071022/37f74686-2d18-49b6-8496-66c13f40c8cb)

TIA

**Comment by bios6** [comment]
Hello, In this case can you change KENDRA_INDEXED_DOCUMENTS_LANGUAGES to be ```fr```  ? Thanks!

**Comment by Guillaume-Bourque-Levio** [comment]
Hello, if a replace the en by fr I get no result.

Can we specify more than 1 language in that parameter ?

Best

**Comment by bios6** [comment]
Hi @Guillaume-Bourque-Levio ,

I'm unable to see any issue with Kendra integration with QnABot. I just tried to deploy an environment with Swedish as the Language in the CloudFormation Parameter and a Kendra Index with Swedish data and I am also able to query in Swedish (with or without the multi-language setting enabled) as well as other languages (if I have multi-language setting enabled ) . Our recommendation would be to turn on your debugging in the settings and as well as try to change around the score threshold for Kendra to see if that helps. 

To answer your previous question KENDRA_INDEXED_DOCUMENTS_LANGUAGES supports a comma separate list. For more information about the settings on QnABot you can check out all parameters and descriptions here: https://github.com/aws-solutions/qnabot-on-aws/blob/main/docs/settings.md . You can also try to download our Implementation guide here https://aws.amazon.com/solutions/implementations/qnabot-on-aws/ which contains a lot of information about this AWS solution. 

<img width="1469" alt="Screenshot 2024-04-16 at 8 43 51 AM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/271434d7-72b8-451b-9901-7fd5dff8c228">
<img width="690" alt="Screenshot 2024-04-16 at 8 48 26 AM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/888674ae-4089-4332-8a6e-e3e7cf671f44">

**Comment by Guillaume-Bourque-Levio** [comment]
Hello @bios6 

could you please share your QnA bot configurations parameters that allow you solution to get the swedish information,

TIA.

**Comment by bios6** [comment]
Hi @Guillaume-Bourque-Levio ,

Here's some images of how I setup my cloudformation parameters and qnabot settings . I only recall touching 2-3 settings mainly for debugging but most are default.

CFN Parameters: 
<img width="696" alt="Screenshot 2024-04-16 at 2 25 14 PM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/92df45c2-b274-4af1-a1b8-2d2806c9f384">
<img width="707" alt="Screenshot 2024-04-16 at 2 26 44 PM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/5e402af0-bad0-4725-814b-98c7462b9356">


QnABot settings:
<img width="1293" alt="Screenshot 2024-04-16 at 2 32 06 PM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/25fd36ff-901c-4474-97fa-5893d846314a">
<img width="1200" alt="Screenshot 2024-04-16 at 2 32 28 PM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/0329e174-e63c-4705-b5f9-eb4a8d0acd77">
<img width="1405" alt="Screenshot 2024-04-16 at 2 33 54 PM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/b234d29d-341a-4481-af40-751ababbc38c">
<img width="1313" alt="Screenshot 2024-04-16 at 2 35 44 PM" src="https://github.com/aws-solutions/qnabot-on-aws/assets/135276684/5a95c9f2-3f04-4087-848b-d856863a7238">


I hope this helps!

**Comment by Guillaume-Bourque-Levio** [comment]
Thanks, I'm now able to get my french answer.

**Comment by fhoueto-amz** [comment]
@Guillaume-Bourque-Levio 
Can you please share what was the root cause of the issue on your side.  @bios6 shared the suggestion on the KENDRA_INDEXED_DOCUMENTS_LANGUAGES  a few comments before but you mentioned that it was still not working then.

**Comment by Guillaume-Bourque-Levio** [comment]
@bios6 can you confirm that in your kendra index you had no English content at all, when you did your test ?

Also was kendra configures from the QnA bot console, or you when into Amazon console and created an index from their ?

We are usinge the crawler v2 extentions to bring data in kendra with the locale set to fr.

We are still having a hard time to have a working solution if we have no english content in Kendra


Best

**Comment by fhoueto-amz** [comment]
Fixed in v6.0.0
